config/routes.rb

```diff
@@ -1,8 +1,10 @@
 Rails.application.routes.draw do
   get '/users', to: 'users#index'
   root 'users#index'
-
+ 
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create]
+    end
```

Comment:
- [ ] As per the requirement, the user should also can add likes to the posts. Please add the likes_controller and the required path here.